staging/src/k8s.io/component-base/cli/flag/sectioned_test.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMergeNamedFlagSets(t *testing.T) {
+	nfs1 := NamedFlagSets{}
+	nfs1.FlagSet("flagSet1")
+
+	nfs2 := NamedFlagSets{}
+	nfs2.FlagSet("flagSet2")
+
+	nfs3 := MergeNamedFlagSets(nfs1, nfs2)
+	assert.Len(t, nfs3.FlagSets, 2)
```

Comment:
I think the assert needs to be more than this.
It needs to not just check the length, but make sure the FlagSets contained within are the expected ones.
Also, `Order` and `NormalizeNameFunc` should probably be checked to make sure they match what is expected here.

Reply:
I've added one assertion for `Order`, I could not find a obvious way to test the `NormalizeNameFunc` so far, I'll have a second look tomorrow.